### PR TITLE
Replaced CHECK_ by TORCH_CHECK_

### PR DESCRIPTION
**Pull request by vfdev-5**
Try to fix failing CI

Related pytorch PR: https://github.com/pytorch/pytorch/pull/82032

CUDA CI jobs are failing as they still install 1.13.0.dev20220726 but CPU CI jobs are already running on 1.13.0.dev20220727
-> Check https://anaconda.org/pytorch-nightly/pytorch/files
-> Restart failing CI jobs only

### CODE REVIEW

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

**Comment by datumbox** [comment]
@vfdev-5 Some test are [failing](https://app.circleci.com/pipelines/github/pytorch/vision/19037/workflows/bcec5838-e8d9-4c3f-81ce-ad40dcc6bbcb/jobs/1540198).

**Comment by vfdev-5** [comment]
Yes, it is related as these jobs are install yesterday's pytorch nightly where `CHECK_` is still there. Let's wait for uploaded binaries and restart failing CI jobs

**Review by datumbox — APPROVED** [review]

File: torchvision/csrc/io/decoder/sync_decoder.cpp [hunk 1]
```diff
@@ -19,17 +19,17 @@ void SyncDecoder::AVByteStorage::ensure(size_t n) {
...
 
 uint8_t* SyncDecoder::AVByteStorage::writableTail() {
-  CHECK_LE(offset_ + length_, capacity_);
+  TORCH_CHECK_LE(offset_ + length_, capacity_);
   return buffer_ + offset_ + length_;
 }
 
 void SyncDecoder::AVByteStorage::append(size_t n) {
-  CHECK_LE(n, tail());
+  TORCH_CHECK_LE(n, tail());
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/sync_decoder.cpp:27]: This is not just a check. This is a function call that must be implemented and then checked for its output. Unfortunately the new `TORCH_CHECK_LE` mechanism can skip completely the method call if not in debug mode. This means that the function call will never be done causing issues.

File: torchvision/csrc/io/decoder/sync_decoder_test.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ void gotFilesStats(std::vector<VideoFileStats>& stats) {
     fseek(f, 0, SEEK_END);
     std::vector<uint8_t> buffer(ftell(f));
     rewind(f);
-    CHECK_EQ(buffer.size(), fread(buffer.data(), 1, buffer.size(), f));
+    TORCH_CHECK_EQ(buffer.size(), fread(buffer.data(), 1, buffer.size(), f));
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/sync_decoder_test.cpp:53]: Another problematic call.

File: torchvision/csrc/io/decoder/sync_decoder_test.cpp [hunk 3]
```diff
@@ -90,7 +90,7 @@ size_t measurePerformanceUs(
...
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/sync_decoder_test.cpp:93]: another one

File: torchvision/csrc/io/decoder/sync_decoder_test.cpp [hunk 4]
```diff
@@ -324,7 +324,7 @@ TEST(SyncDecoder, TestMemoryBuffer) {
   fseek(f, 0, SEEK_END);
   std::vector<uint8_t> buffer(ftell(f));
   rewind(f);
-  CHECK_EQ(buffer.size(), fread(buffer.data(), 1, buffer.size(), f));
+  TORCH_CHECK_EQ(buffer.size(), fread(buffer.data(), 1, buffer.size(), f));
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/sync_decoder_test.cpp:327]: this one too

File: torchvision/csrc/io/decoder/sync_decoder_test.cpp [hunk 5]
```diff
@@ -349,7 +349,7 @@ TEST(SyncDecoder, TestMemoryBufferNoSeekableWithFullRead) {
...
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/sync_decoder_test.cpp:352]: this one too

File: torchvision/csrc/io/decoder/sync_decoder_test.cpp [hunk 6]
```diff
@@ -388,7 +388,7 @@ TEST(SyncDecoder, TestMemoryBufferNoSeekableWithPartialRead) {
...
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/sync_decoder_test.cpp:391]: this one too

File: torchvision/csrc/io/decoder/util.cpp [hunk 7]
```diff
@@ -265,7 +265,7 @@ std::string generateErrorDesc(int errorCode) {
 
 size_t serialize(const AVSubtitle& sub, ByteStorage* out) {
   const auto len = size(sub);
-  CHECK_LE(len, out->tail());
+  TORCH_CHECK_LE(len, out->tail());
```
Comment (datumbox) [inline torchvision/csrc/io/decoder/util.cpp:268]: this one too

**Review by datumbox — COMMENTED** [review]
Here is what TORCH_CHECK_* are defined:
```C++

#include <glog/logging.h>

// Additional macros on top of glog
#ifndef NDEBUG
#define TORCH_CHECK_EQ(val1, val2) CHECK_EQ(val1, val2)
#else // !NDEBUG
// These versions generate no code in optimized mode.
#define TORCH_CHECK_EQ(val1, val2) \
  while (false)                    \
  CHECK_EQ(val1, val2)
#endif // NDEBUG
```


This PR broke FB internals. It seems that the real cause of the problem is the change on the semantics/behaviour of `*CHECK*` on Core. See https://github.com/pytorch/pytorch/pull/82032#issuecomment-1203695164.

Regardless, to resolve the situation I'm issuing a forward fix at [D38383266](https://www.internalfb.com/diff/D38383266).

**Pull request merged**